app/controllers/postgresql_lo_streamer/lo_controller.rb

```diff
@@ -16,6 +23,18 @@ def stream
       end
     end
 
+    def object_exists?(identifier)
+      begin
+        connection.lo_open(identifier, ::PG::INV_READ)
+      rescue PG::Error => e
+        if e.to_s.include? "does not exist"
```

Comment:
In case this if evaluates to false `object_exists?` will return true even if the `lo_open` failed. I believe it would be better to re-raise the exception in that case.